ts-client/src/amm/index.ts

```diff
@@ -567,6 +574,17 @@ export default class AmmImpl implements AmmImplementation {
     }
 
     const [mintMetadata, _mintMetadataBump] = deriveMintMetadata(lpMint);
+    const lpAmount = sqrt(tokenAAmount.mul(tokenBAmount));
+    const feeWrapperPercent = opt?.stakeLiquidity?.percent?.gt(new Decimal(0))
+      ? opt.stakeLiquidity.percent
+      : new Decimal(0);
+
+    const feeWrapperLockAmount = new BN(
+      new Decimal(lpAmount.toString()).mul(feeWrapperPercent).toFixed(0, Decimal.ROUND_DOWN),
+    );
+    const userLockAmount = feeWrapperPercent.gt(new Decimal(0)) ? new BN(
```

Comment:
Should be `userLockAmount = feeWrapperPercent.gt(new Decimal(0)) ? lpAmount.sub(feeWrapperLockAmount) : U64_MAX`
If not, there's some precision loss will be remained in the user wallet.
Example:
```
lpAmount = 133
feeFarmLockPct = 0.1
userLockPct = 0.9

feeWrapperLockAmount = 133 * 0.1 = RoundDown(13.3) = 13
userLockAmount = 133 * (1 - 0.1) = RoundDown(119.7) = 119
119 + 13 != 133
```